src/libAtomVM/context.h

```diff
@@ -244,49 +239,57 @@ static inline void context_clean_registers(Context *ctx, int live)
 }
 
 /**
- * @brief Returns available free memory in term units
+ * @brief Returns a context's stack base
  *
- * @details Returns the number of terms that can fit either on the stack or on the heap.
+ * @details Used for stack traces
  * @param ctx a valid context.
- * @returns available free memory that is avail_size_in_bytes / sizeof(term).
+ * @returns the stack base
  */
-static inline unsigned long context_avail_free_memory(const Context *ctx)
+static inline term *context_stack_base(const Context *ctx)
 {
-    return ctx->e - ctx->heap_ptr;
-}
-
-/**
- * @brief Returns context total memory in term units
- *
- * @details Returns the total memory reserved for stack and heap in term units.
- * @param ctx a valid context.
- * @returns total memory that is total_size_in_bytes / sizeof(term).
- */
-static inline unsigned long context_memory_size(const Context *ctx)
-{
-    return ctx->stack_base - ctx->heap_start;
+    // Find which fragment the stack belongs to.
+    if (ctx->e >= ctx->heap.heap_start && ctx->e <= ctx->heap.heap_end) {
+        return ctx->heap.heap_end;
+    }
+    HeapFragment *fragment = ctx->heap.next;
+    while (fragment) {
```

Comment:
I prefer fragment != NULL.  I know in the olden days there was no guarantee in the C standard that NULL == 0 (unlike C++).  Maybe that has changed in C, but my habit hasn't.  Take it as you like.  Not a deal breaker.  Just hits my eye when I read the code.

Reply:
I have yet to find which version of C standard didn't accept that :-)

https://c-faq.com/null/ptrtest.html
https://c-faq.com/null/nullor0.html

Reply:
:ack: